swarm/network/stream/delivery.go

```diff
@@ -181,12 +182,23 @@ func (d *Delivery) handleRetrieveRequestMsg(ctx context.Context, sp *Peer, req *
 	return nil
 }
 
+//Chunk delivery always uses the same message type....
 type ChunkDeliveryMsg struct {
-	Addr  storage.Address
-	SData []byte // the stored chunk Data (incl size)
-	peer  *Peer  // set in handleChunkDeliveryMsg
+	Addr    storage.Address
+	SData   []byte // the stored chunk Data (incl size)
+	peer    *Peer  // set in handleChunkDeliveryMsg
+	Syncing bool   // if true, this is a delivery for syncing (no SWAP accounting needed)
```

Comment:
this would be needed related to your next comment - if `peer.Deliver()` is called from two different places, then it's needed. But if we create two different `Delivery` methods, it would not be needed.

I think it is cheapest to use this flag. The implementation logic is in fact the same. If we have major refactorings, then having two different methods will require double work. I vote to keep this flag.

Reply:
ok